src/charm.py

```diff
@@ -173,12 +215,81 @@ def __init__(self, *args):
 
         self._set_status()
 
+    def _on_cos_agent_relation_joined(self, _):
+        self._stored.have_gagent = True
+        self._set_status()
+
+    def _on_cos_agent_relation_broken(self, _):
+        self._stored.have_gagent = False
+        self._set_status()
+
+    def _delete_existing_dashboard_files(self, dashboards_dir: str):
+        directory = Path(dashboards_dir)
+        if directory.exists():
+            for file_path in directory.glob("request_*.json"):
+                file_path.unlink()
```

Comment:
I wonder if we shouldn't log something on debug level for these file and directory operations and handle filesystem exceptions. Although filesystem failures aren't too likely, they sometimes happen. If the dashboards aren't created or deleted it likely doesn't mean yet COS proxy isn't functional but the operator might need to be able to find what happened.